jax/interpreters/xla.py

```diff
@@ -122,11 +122,14 @@ def array_result_handler(device: Optional[Device], aval: core.ShapedArray):
 }
 
 def device_put(x, device: Optional[Device] = None) -> Tuple[Any]:
-  x = canonicalize_dtype(x)
-  try:
-    return device_put_handlers[type(x)](x, device)
-  except KeyError as err:
-    raise TypeError(f"No device_put handler for type: {type(x)}") from err
+  handler = device_put_handlers.get(type(x))
+  if handler:
+    x = canonicalize_dtype(x)
```

Comment:
It looks like this PR broke internal tests due to this line. In particular, some type like `IntEnum` no longer are accepted by `device_put`.

Reply:
Oh man, I think you nailed it! Is it just because I dropped the `x = canonicalize_dtype(x)` beforehand?

Reply:
That was the only thing I saw that could have broken the existing code path.